### ISSUE TITLE
GPS not working

### ISSUE THREAD

**Issue by Iamyahodfpv** [Qrome/QLiteOSD#7]
I currently have a foxeer m10 wired up and am not getting any gps info.

**Comment by Qrome** [comment]
Are you seeing any of the other OSD elements?

**Comment by Iamyahodfpv** [comment]
Yes, just not getting info from the gps

**Comment by Iamyahodfpv** [comment]
The gps elements are there but no values show up

**Comment by Iamyahodfpv** [comment]
The gps is a m10q 250, is there something I need to change in the config?

**Comment by Iamyahodfpv** [comment]
Also the baud rate is 115200

**Comment by Qrome** [comment]
You can change the baud rate in the firmware and compile it on your own.   The default is expecting 9600.   If you need me to compile it, let me know.

**Comment by Iamyahodfpv** [comment]
What tab would that be under in the arduino io?

**Comment by Qrome** [comment]
It is part of the QLiteOSD.ino code.   Near line 58.

`static const uint32_t GPSBaud = 9600;`

Change 9600 to 115200 and compile it.   If you are not able to compile it, let me know the options you want and I can do it for you.

**Comment by Iamyahodfpv** [comment]
Well when I plugged the board back in I didn’t have the pins lined up and I think I may have bricked it. Is there a way to fix that?

**Comment by Iamyahodfpv** [comment]
Now I’m just getting a solid blue light and no osd

**Comment by Qrome** [comment]
Yeah, reversing the plugs can toast the Wemos D1 Mini board -- good news is you can replace those.

**Comment by Iamyahodfpv** [comment]
Is it worth trying to uninstall and reinstall the drivers? It doesn’t smell like I smoked anything.

**Comment by Iamyahodfpv** [comment]
I didn’t plug it in the wrong way just miss aligned the pins by one slot

**Comment by Qrome** [comment]
I have seen it toast the Wemos D1 mini if the pins get plugged in wrong.   Solid light or not light is an indicator.   Remove the Wemos D1 mini from the socket and try to run it off the USB cable and or load it again -- if it will not load or if the LED does not start flashing then it got toasted.   They are easy to replace.

**Comment by Iamyahodfpv** [comment]
Yeah not getting anything when plugged into usb, it shows up but won’t connect

**Comment by Qrome** [comment]
Yeah, sorry to hear that.

**Comment by Iamyahodfpv** [comment]
Okay so I got the new board, flashed it but when I plug it in I'm not getting anything. No flashing led when plugged into the board but it does flash when plugged into my computer

**Comment by Qrome** [comment]
That's an indicator that there might be a bad solder joint or joined pins.  Did you solder the d3 pin and gnd for wifi mode?  If so, make sure that is correct.

**Comment by Qrome** [comment]
Post some photos of the board.

**Comment by Iamyahodfpv** [comment]
It was the button for the WiFi, removed it and now it works. But still am not getting gps info

**Comment by Iamyahodfpv** [comment]
![image](https://github.com/Qrome/QLiteOSD/assets/101999008/75785ada-aff3-4a63-b5a7-a886ab0e8dda)

**Comment by Iamyahodfpv** [comment]
I changed the baud rate but still no gps data in the osd. I wired up the gps to a quad to make sure it wasn't the gps and it worked fine on that

**Comment by Qrome** [comment]
Make sure you have the TX to RX pins correct. Try to swap them.  Also, you may need to take it outside to get satellites.

**Comment by Iamyahodfpv** [comment]
Rx is going to tx and tx to rx

**Comment by Iamyahodfpv** [comment]
![image](https://github.com/Qrome/QLiteOSD/assets/101999008/60a2b6d4-92e5-4be0-9054-ac85fcab4e5b)

**Comment by Iamyahodfpv** [comment]
It's weird that it is showing 14 sats but no longitude or latitude

**Comment by Qrome** [comment]
If it is flashing 14 then it is actually no satellites.  Try it outside... Satellite link may require you to have it outside.

**Comment by Iamyahodfpv** [comment]
The gps is blinking green, which means it has a lock. I'll try switching around rx and rx but they are wired the way they should be

**Comment by Iamyahodfpv** [comment]
*rx and tx

**Comment by Qrome** [comment]
Then, maybe that GPS is not compatible.   If it has a compass included that might be part of the problem.

**Comment by Iamyahodfpv** [comment]
Would there be any changes I need to make to the code for an m10 without a compass?

**Comment by Qrome** [comment]
The project is using the TinyGPS++ library.   You may want to check over on that project for compatibility.
